phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -634,12 +635,24 @@ public void preBatchMutate(ObserverContext<RegionCoprocessorEnvironment> c,
           final Set<String> haGroupNames = extractHAGroupNameAttribute(miniBatchOp);
           // Check if mutation is blocked for any of the HAGroupNames
           for (String haGroupName : haGroupNames) {
-              if (StringUtils.isNotBlank(haGroupName)
-                      && haGroupStoreManager.isMutationBlocked(haGroupName)) {
-                  throw new MutationBlockedIOException("Blocking Mutation as Some CRRs are in "
-                          + "ACTIVE_TO_STANDBY state and "
-                          + "CLUSTER_ROLE_BASED_MUTATION_BLOCK_ENABLED is true");
-              }
+                //TODO: Below approach might be slow need to figure out faster way, slower part is
+                //getting haGroupStoreClient We can also cache roleRecord (I tried it and still its
+                //slow due to haGroupStoreClient initialization) and caching will give us old result
+                //in case one cluster is unreachable instead of UNKNOWN. Check if mutation's
+                //haGroup is stale
+                boolean isHAGroupOnClientStale = haGroupStoreManager
```

Comment:
If there is a local uncertainty about state, either because state is missing or not initialized yet or if the remote gave back an error suggesting client understanding is not current, refresh the CRR, then decide what to do next. I think this is the way.